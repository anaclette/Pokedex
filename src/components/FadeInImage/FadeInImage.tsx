import React, {useState} from 'react';
import {
  ActivityIndicator,
  Animated,
  ImageStyle,
  StyleProp,
  View,
  ViewStyle,
} from 'react-native';
import {useAnimation} from '../../hooks/useAnimation';
import colors from '../../themes/colors';
import {styles} from './fadeInImage.style';

interface Props {
  uri: string;
  imgStyle?: StyleProp<ImageStyle>;
  containerStyle?: StyleProp<ViewStyle>;
}

export const FadeInImage = ({
  uri,
  imgStyle = {},
  containerStyle = {},
}: Props) => {
  const {opacity, fadeIn} = useAnimation();
  const [isLoading, setIsLoading] = useState(true);

  const finishLoading = () => {
    setIsLoading(false);
    fadeIn();
  };

  const onError = () => setIsLoading(false);

  return (
    <View style={containerStyle ? containerStyle : styles.container}>
      {isLoading && (
        <ActivityIndicator
          style={styles.loader}
          color={colors.dark}
          size={30}
        />
      )}

      <Animated.Image
        source={{uri}}
        onError={onError}
        onLoad={finishLoading}
        style={{
          ...(imgStyle as ImageStyle),
          opacity,
        }}
      />
    </View>
  );
};
