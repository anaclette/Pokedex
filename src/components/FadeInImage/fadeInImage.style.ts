import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  loader: {
    flex: 1,
    top: '20%',
  },
});
