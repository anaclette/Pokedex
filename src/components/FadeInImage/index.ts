import {FadeInImage} from './FadeInImage';

export default FadeInImage;
