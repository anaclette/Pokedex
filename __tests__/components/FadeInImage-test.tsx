import React from 'react';
import {render} from '@testing-library/react-native';
import FadeInImage from '../../src/components/FadeInImage';
import renderer from 'react-test-renderer';

describe('Testing FadeInImage component', () => {
  it('Should render FadeInImage correctly', () => {
    const component = <FadeInImage uri={'uri source'} />;
    const {getByTestId} = render(component);

    expect(renderer.create(component).toJSON()).toMatchSnapshot();
    expect(getByTestId('image')).toBeTruthy();
    expect(getByTestId('activity-indicator')).toBeTruthy();

    // const handleLoading = jest.spyOn(React, 'useState');
    // const setIsLoading = jest.fn();
    // handleLoading.mockImplementation(loading => [loading, setIsLoading]);
    // expect(setIsLoading).toBeCalled();
  });
});
