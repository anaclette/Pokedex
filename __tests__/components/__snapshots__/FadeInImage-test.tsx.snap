// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing FadeInImage component Should render FadeInImage correctly 1`] = `
<View
  style={Object {}}
>
  <ActivityIndicator
    color="#291720"
    size={30}
    style={
      Object {
        "flex": 1,
        "top": "20%",
      }
    }
    testID="activity-indicator"
  />
  <Image
    accessibilityRole="image"
    collapsable={false}
    onError={[Function]}
    onLoad={[Function]}
    source={
      Object {
        "uri": "uri source",
      }
    }
    style={
      Object {
        "opacity": 0,
      }
    }
    testID="image"
  />
</View>
`;
